platform/os_windows.go

```diff
@@ -189,6 +203,96 @@ func SetSdnRemoteArpMacAddress() error {
+func SetMellanoxRegKeyPriorityVLANTag() error {
+	//get mellanox adapter name
+	cmd := fmt.Sprintf("Get-NetAdapter | Where-Object { $_.InterfaceDescription -like \"%s\" } | Select-Object -ExpandProperty Name", mellanoxSearchString)
+	adapatername, err := ExecutePowershellCommand(cmd)
+
+	if err != nil {
+		return fmt.Errorf("error while executing powershell command to get net adapter list: %w", err)
+	}
+	if adapatername == "" {
+		return fmt.Errorf("no network adapter found with \"%s\" in description", mellanoxSearchString)
+	}
+
+	log.Printf("Name of Mellanox adapter : %v", adapatername)
+
+	//Find if adapter has property PriorityVLANTag (version 4 or up) or not (version 3)
+	cmd = fmt.Sprintf("Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty Name", priorityVLANTagIdentifier, adapatername)
+	adapterNamewithVLANTag, err := ExecutePowershellCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error while executing powershell command to get VLAN Tag advance property of %s: %w", adapatername, err)
+	}
+
+	if adapterNamewithVLANTag != "" {
+		//adapter is version 4 and up since adapter's advance property consists of reg key : PriorityVLANTag
+		//set reg value for Priorityvlantag of adapater to 3
+		cmd = fmt.Sprintf("Set-NetAdapterAdvancedProperty -Name \"%s\" -RegistryKeyword \"%s\" -RegistryValue %d", adapatername, priorityVLANTagIdentifier, regvalueForVLANTag)
+		_, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while setting up registery value for PriorityVLANTag for adapater: %w", err)
+		}
+		//check if value is set or not
+		cmd = fmt.Sprintf(" Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty RegistryValue", priorityVLANTagIdentifier, adapatername)
+		regvalue, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while checking registery value for PriorityVLANTag for adapater: %w", err)
+		}
+		if regvalue != fmt.Sprintf("%d", regvalueForVLANTag) {
+			return fmt.Errorf("failure while setting Mellanox Network Adapter: \"%s\" with \"%s\" property value to %d", adapatername, priorityVLANTagIdentifier, regvalueForVLANTag)
+		}
+		log.Printf("Successfully set Mellanox Network Adapter: \"%s\" with \"%s\" property value as %d", adapatername, priorityVLANTagIdentifier, regvalueForVLANTag)
+	} else {
+		//Adapter is version 3 or less as PriorityVLANTag was not found in advanced properties of mellanox adpater
+		log.Printf("Searching through CIM instances for Network devices with \"%s\" in the name", mellanoxSearchString)
+		cmd = fmt.Sprintf("Get-CimInstance -Namespace root/cimv2 -ClassName Win32_PNPEntity | Where-Object PNPClass -EQ \"Net\" | Where-Object { $_.Name -like \"%s\" } | Select-Object -ExpandProperty DeviceID", mellanoxSearchString)
+		deviceid, err := ExecutePowershellCommand(cmd)
+
```

Comment:
can you look at other files for consistent coding style? I see in some places you have spaces before if, in other places, there are no spaces

Reply:
sure will do